### PR TITLE
Re-enable format checking and reformat all files with findings.

### LINKED ISSUE #815: Improve Nighthawk's check_format.py
Nighthawk's [check_format.py](https://github.com/envoyproxy/nighthawk/blob/d3d6d3a60ce02ea9cf0466f16534f89fdd6a2c58/tools/check_format.sh#L1) is reusing [the Envoy version](https://github.com/envoyproxy/envoy/blob/d3bd9bbef64abbf1c02fe91231eac8841901131d/tools/code_format/check_format.py#L1).

Envoy includes a set of Envoy specific checks that cannot be applied in the Nighthawk codebase without code changes upstream. As an example, the tool [checks](https://github.com/envoyproxy/envoy/blob/d3bd9bbef64abbf1c02fe91231eac8841901131d/tools/code_format/check_format.py#L746-L750) that the `TRY_ASSERT_MAIN_THREAD` is used instead of a raw `try`, yet the `TRY_ASSERT_MAIN_THREAD` cannot be used in Nighthawk's code in its current form due to namespace issues.

Nighthawk was getting around this by running `check_format.py` with an absolute path, so [checks like this](https://github.com/envoyproxy/envoy/blob/d3bd9bbef64abbf1c02fe91231eac8841901131d/tools/code_format/check_format.py#L746-L747) excluded these Envoy specific verifications. This however was a dependence on an undocumented behavior r even a bug.

Envoy modified their `check_format.py` in https://github.com/envoyproxy/envoy/pull/20108, making it impossible to execute with absolute paths.

As a workaround we are now executing the script with a path like `./../nighthawk` which again allows bypassing of these Envoy specific checks.

We should find a permanent solutions, which could be either of:
1. standardize a way to exclude checks in Envoy's `check_format.py`; or
2. modify code as needed to allow Nighthawk to fully comply with the Envoy specific checks.

### PR DESCRIPTION
**Pull request by mum4k**
Turns out our file format checkers stopped working some time ago (likely when the `--path` argument was added here: https://github.com/envoyproxy/envoy/pull/29397). Once the path was set, the formatting tools executed, but they were checking format of Envoy files instead of Nighthawk files.

Most formatting changes were done automatically except:
- addition of missing Python docstrings.
- replacement of references to `google::protobuf` by references to the `Envoy::Protobuf` namespace.

Fixes #815

### CODE REVIEW

File: include/nighthawk/adaptive_load/input_variable_setter.h [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#pragma once
```
Comment (chipmunkie) [inline include/nighthawk/adaptive_load/input_variable_setter.h:1]: This already existed after the comment.

File: include/nighthawk/adaptive_load/metrics_plugin.h [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#pragma once
```
Comment (chipmunkie) [inline include/nighthawk/adaptive_load/metrics_plugin.h:1]: As above, the second #pragma once needs to be removed.

File: include/nighthawk/adaptive_load/step_controller.h [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#pragma once
+
 // Interfaces for StepController plugins and plugin factories.
 
 #pragma once
```
Comment (chipmunkie) [inline include/nighthawk/adaptive_load/step_controller.h:5]: please remove.

File: source/client/flush_worker_impl.h [hunk 4]
```diff
@@ -1,3 +1,5 @@
+#pragma once
+
 // Flush worker implementation. Flush worker periodically flushes metrics
 // snapshot to all configured stats sinks in Nighthawk.
 #pragma once
```
Comment (chipmunkie) [inline source/client/flush_worker_impl.h:5]: Please remove.

File: source/request_source/request_options_list_plugin_impl.h [hunk 5]
```diff
@@ -1,3 +1,5 @@
+#pragma once
+
 // Implementations of RequestSourceConfigFactories that make a OptionsListRequestSource.
 #pragma once
```
Comment (chipmunkie) [inline source/request_source/request_options_list_plugin_impl.h:4]: Please remove.

File: test/request_source/stub_plugin_impl.h [hunk 6]
```diff
@@ -1,3 +1,5 @@
+#pragma once
+
 // Test implementations of RequestSourceConfigFactory and RequestSource that perform minimum
 // functionality for testing purposes.
 #pragma once
```
Comment (chipmunkie) [inline test/request_source/stub_plugin_impl.h:5]: Please remove.

**Review by chipmunkie — COMMENTED** [review]
I only found a  few nits.

File: include/nighthawk/adaptive_load/input_variable_setter.h (hunk 1 above)
Comment (mum4k) [reply include/nighthawk/adaptive_load/input_variable_setter.h:1]: Thank you for spotting these. Done in 78338dca6de640193a4ea395ed2052c832397ea8.

File: include/nighthawk/adaptive_load/metrics_plugin.h (hunk 2 above)
Comment (mum4k) [reply include/nighthawk/adaptive_load/metrics_plugin.h:1]: Done in 78338dca6de640193a4ea395ed2052c832397ea8.

File: include/nighthawk/adaptive_load/step_controller.h (hunk 3 above)
Comment (mum4k) [reply include/nighthawk/adaptive_load/step_controller.h]: Done in 78338dca6de640193a4ea395ed2052c832397ea8.

File: source/client/flush_worker_impl.h (hunk 4 above)
Comment (mum4k) [reply source/client/flush_worker_impl.h]: Done in 78338dca6de640193a4ea395ed2052c832397ea8.

File: source/request_source/request_options_list_plugin_impl.h (hunk 5 above)
Comment (mum4k) [reply source/request_source/request_options_list_plugin_impl.h]: Done in 78338dca6de640193a4ea395ed2052c832397ea8.

File: source/adaptive_load/scoring_function_impl.h [hunk 7]
```diff
@@ -1,3 +1,5 @@
+#pragma once
+
 // Implementations of ScoringFunction plugins and corresponding factories.
 #pragma once
```
[the remark this answers is not in the document]
Comment (mum4k) [reply source/adaptive_load/scoring_function_impl.h]: Done in 78338dca6de640193a4ea395ed2052c832397ea8.

**Review by chipmunkie — APPROVED** [review]
LGTM

**Pull request merged**